### PR TITLE
Logger

### PR DESCRIPTION
**Pull request by uint-jamf**
image

image

I chose this one: https://docs.rs/env_logger/0.7.1/env_logger/

It lets us do nice defaults and it's easy to override them using env vars.

Note in the first screenshot you can see a dev and then a release build with different default log levels.

### CODE REVIEW

File: src/bin/main.rs [hunk 1]
```diff
@@ -4,19 +4,38 @@ use std::result::Result;
...
-        eprintln!("Error: {}", err);
+        error!("{}", err);
 
         std::process::exit(1);
-    }
+    };
+}
+
+/// Initializes the logger. It'll be more verbose by default in dev builds and
+/// more "tidy" in releases. It can be customized via env variables.
```
[the remark this answers is not in the document]
Comment (thecodesmith) [reply src/bin/main.rs:27]: And if they are logger-specific variables, could we have our own, like `RENDR_LOG=info`?

File: src/blueprint.rs [hunk 2]
```diff
@@ -132,7 +132,7 @@ impl Blueprint {
             let output_path = output_dir.join(file.path_from_template_root());
 
             if path.is_file() {
-                println!("Found file {:?}", &path);
+                info!("Found file {:?}", &path);
```
[the remark this answers is not in the document]
Comment (uint-jamf) [reply src/blueprint.rs:135]: Oh, sure. Actually, maybe we should make all of those debug?

File: src/bin/main.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (uint-jamf) [reply src/bin/main.rs:27]: They're logger-specific and look like this: `RUST_LOG=error rendr ...`. More info is here: https://docs.rs/env_logger/0.7.1/env_logger/

It looks like we could change it to `RENDR_LOG`. Do we want to?

**Comment by uint-jamf** [comment]
How's this?

Dev build:
image

Release build:
image

**Comment by uint-jamf** [comment]
This should be good for another look. I tried to keep the logic as simple as possible there since it... weirds me out if this is too complex.

In release builds, if the env var `RENDR_LOG` exists, it will print the logging prefix. Otherwise it won't.

**Pull request merged**